lib/bojangles.rb

```diff
@@ -87,25 +87,36 @@ def cached_error_messages
     end
   end
 
+  # Only the first two lines of an error message need to match for us to decide
+  # that we've already sent it.
+  def compare_errors(current_errors, old_errors)
+    new_errors = current_errors.reject do |error|
```

Comment:
Subtraction's really fine when you're doing simple comparisons, but for something more sophisticated like this you need a block method.